src/thunderbots/software/grsim_communication/motion_controller/motion_controller.cpp

```diff
@@ -84,29 +85,21 @@ Vector MotionController::determineLinearVelocity(const Robot robot, const Point
     double max_magnitude_of_velocity_to_apply =
         ROBOT_MAX_ACCELERATION_METERS_PER_SECOND_SQUARED * delta_time;
 
-    // Calculate a vector from the robot to the destination
-    Vector vector_to_dest = dest - robot.position();
-
-    // Calculate the component of our current velocity pointing towards the destination
-    double magnitude_of_robot_velocity_towards_dest =
-        robot.velocity().len() *
-        (vector_to_dest.orientation() - robot.velocity().orientation()).cos();
-    Vector robot_velocity_towards_dest =
-        vector_to_dest.norm(magnitude_of_robot_velocity_towards_dest);
+    // Calculate a unit vector from the robot to the destination
+    Vector unit_vector_to_dest = (dest - robot.position()).norm();
 
-    // Calculate the vector required to counteract our movement perpendicular to the
-    // vector towards the dest. This "pushes" us onto a velocity vector straight towards
-    // the destination
+    // Calculate the vector of our velocity perpendicular to the vector toward the
+    // destination
     Vector robot_velocity_perpendicular_to_dest =
-        robot.velocity() - robot_velocity_towards_dest;
+        robot.velocity().dot(unit_vector_to_dest.perp()) * unit_vector_to_dest.perp();
```

Comment:
What's up with the last `* unit_vector_to_dest.perp();` here? I understand the first bit, but after that you should have the perpendicular component of the motion, no?

Reply:
the dot product is the length of the projection of `robot.velocity()` onto `unit_vector_to_dest.perp()`, and multiplying by `unit_vector_to_dest.perp()` gives the vector that is the image of `robot.velocity()` onto `unit_vector_to_dest.perp()`, which is what we are looking for.

Reply:
ah of course. whoops!